src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2863,6 +2864,12 @@ def transform(dataset):
                 if method:
                     dataset[i]['value'] = method(dataset[i]['value'])
 
+            if 'mountpoint' in dataset:
+                # This is treated specially to keep backwards compatibility with API
+                dataset['mountpoint'] = dataset['mountpoint']['value']
```

Comment:
I think mountpoint should always be there, we cannot have top level properties that may or may not be there, its very misleading and backward incompatible.

Reply:
So if mountpoint property was not retrieved explicitly, what should we say for `mountpoint` property ?

Reply:
Ability to specify custom properties was introduced recently - which means that for old users/releases, `mountpoint` property would continue to exist always

Reply:
Recently as in SCALE? not 12?

Reply:
@william-gr, yes that is accurate - for `pool.dataset.query`, retrieving specific properties was introduced in SCALE

Reply:
Ok, then sounds reasonable, thanks for clarifying!